libs/gisday/platform/ngx/data-access/src/lib/modules/services/_base/base.service.ts

```diff
@@ -1,47 +1,86 @@
-import { HttpClient } from '@angular/common/http';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+
+import { OidcSecurityService } from 'angular-auth-oidc-client';
+import { DeepPartial } from 'typeorm';
 
 import { EnvironmentService } from '@tamu-gisc/common/ngx/environment';
 
 export abstract class BaseService<T> {
-  public withCredentials = false;
   public resource: string;
+  public accessToken;
+  public headers: HttpHeaders;
 
-  constructor(private env: EnvironmentService, private http: HttpClient, private route: string, withCreds?: boolean) {
+  constructor(
+    private env: EnvironmentService,
+    private http: HttpClient,
+    public oidcSecurityService: OidcSecurityService,
+    private route: string
+  ) {
     this.resource = this.env.value('api_url') + `/${route}`;
-    this.withCredentials = withCreds;
-  }
 
-  public getEntity(guid: string) {
-    return this.http.get<Partial<T>>(`${this.resource}/${guid}`, {
-      withCredentials: this.withCredentials
+    this.accessToken = this.oidcSecurityService.getAccessToken();
```

Comment:
More elegant solution would have been to use an HttpInterceptor to apply the access token. This removes the need to remember to add it requests. Decouple the service and the authorization. 

We can leave this point for a later PR.